pkg/downloader/downloader.go

```diff
@@ -385,3 +392,86 @@ func downloadHTTP(localPath, url string, expectedDigest digest.Digest) error {
 
 	return nil
 }
+
+// Read the file and return its contents as a string
+// Never use it with large files
+func readSmallTextFile(filename string) (string, error) {
+	data, err := os.ReadFile(filename)
+	if err != nil {
+		return "", err
+	}
+	return string(data), nil
+}
+
+// Read first *.digest file found and return the Digest object from its contents
+func guessDigestFromDir(dir string) (digest.Digest, error) {
+	dirList, err := os.ReadDir(dir)
+	if err != nil {
+		return "", err
+	}
+
+	for _, f := range dirList {
+		if strings.HasSuffix(f.Name(), ".digest") && f.Type().IsRegular() {
```

Comment:
Why only regular?

Reply:
Well.. mainly because caching code stores it as a regular file. I can replace with `!( f.IsDir() )` if you'd prefer though.